stdlib/public/core/SmallString.swift

```diff
@@ -86,7 +108,7 @@ extension _SmallString {
   @inlinable
   internal var isASCII: Bool {
     @inline(__always) get {
-      return _StringObject(rawUncheckedValue: self.rawBits).smallIsASCII
+      return discriminator.smallIsASCII
```

Comment:
I see not much of a difference in generated arm64 code before/after this patch. In fact, the difference is so inconsequential that I suspect I'm not looking at the right thing!

I looked at the generated code for `s.count` and `s1 == s2`. I see some `tb[n]z`s in the `==`, and it looks like this patch doesn't affect them. `s.count` is slightly different, but I can't judge if it's meaningfully worse.

Reply:
Maybe something like:
```swift
@inline(never) public func foo() { print("foo") }
@inline(never) public func bar() { print("bar") }
public func decide(_ s: String) {
  if s._guts.isFastUTF8 { foo() } else { bar() }
}
public func smolDecision(_ s: String) {
  if s._guts.isSmall { foo() } else { bar() }
}
```

For me, with `xcrun <builddir>/bin/swiftc -frontend -O -target arm64-apple-ios12 -emit-object -disable-access-control foo.swift`, produces `tbnz`:

```
_$s3foo6decideyySSF:
	.cfi_startproc
	tbnz	x1, #60, LBB4_2
	b	_$s3fooAAyyF
LBB4_2:
	b	_$s3foo3baryyF
	.cfi_endproc

_$s3foo12smolDecisionyySSF:
	.cfi_startproc
	tbnz	x1, #61, LBB5_2
	b	_$s3foo3baryyF
LBB5_2:
	b	_$s3fooAAyyF
	.cfi_endproc
```

Nice to have the constructors for StringObject too, but these two queries are very important.

Reply:
Nice, thank you! I’ll double check to be sure, but these two are unlikely to be affected, because they are implemented on StringObject, and I left that with the original code on 64-bit platforms.